README.md
# flutter_chat_page_design
Create chat page design with only pass the message list.




## Features

- Chat bubble design.
- change background and text color for sender and receiver.
- Enable or disable grouping by date.
- Other features are in pipeline coming soon.


## Getting started

```dart

Install
  flutter_chat_page_design : 0.0.4

Import

  import 'package:flutter_chat_page_design/flutter_chat.dart';

```

## Usage

You can also look example application.

```dart
ChatListview(
        chatList: chatList,
        isDategroup: true,
        isReverse: true,
        gropByTextsize: 12,
        bubbleBorder: 15,
        senderBgColor: Colors.red,
        receiverBgColor: Colors.green,
        senderTextColor: Colors.yellow,
        receiverTextColor: Colors.purple,

      ),
```
## Images
![Screenshot_2023-08-24-17-31-10-64_082aea295e0e2b19157fadadca43d2cc 1](https://github.com/harshil023/flutter_chat_page_design/assets/50265690/43c6e4ef-9586-42a2-9095-55f6be441b43)

##  Bugs/Requests
If you encounter any problems feel free to open an issue. If you feel the library is missing a feature, please raise a ticket on Github and I'll look into it. Pull request are also welcome.

##  Additional Information
Developer : Harshil Thakkar
https://github.com/harshil023

CHANGELOG.md
## 0.0.4

- Dart Warning Resolved.


lib/flutter_chat.dart
library flutter_chat_ui;

import 'package:flutter/material.dart';
import 'package:flutter_chat_page_design/widget/chatbubble.dart';

import 'model/chatmodel.dart';

class ChatListview extends StatelessWidget {
  const ChatListview(
      {super.key,
      required this.chatList,
      required this.isDategroup,
      this.gropByTextsize = 12,
      this.bubbleBorder = 15,
      this.senderBgColor = Colors.blue,
      this.receiverBgColor = Colors.grey,
      this.senderTextColor = Colors.white,
      this.groupByTextColor = Colors.white,
      this.receiverTextColor = Colors.black,
      this.isReverse = true});

  final List<ChatModel> chatList;
  final bool isDategroup;
  final double gropByTextsize;
  final double bubbleBorder;

  final Color senderBgColor,
      receiverBgColor,
      senderTextColor,
      groupByTextColor,
      receiverTextColor;
  final bool isReverse;

  @override
  Widget build(BuildContext context) {
    return ListView.builder(
        padding: const EdgeInsets.all(0),
        reverse: isReverse,
        itemCount: chatList.length,
        shrinkWrap: true,
        scrollDirection: Axis.vertical,
        itemBuilder: (BuildContext buildContext, int index) {
          return Column(
            crossAxisAlignment: CrossAxisAlignment.center,
            children: [
              Visibility(
                visible: isDategroup
                    ? index != 0
                        ? chatList[index].date == chatList[index - 1].date
                            ? false
                            : true
                        : true
                    : false,
                child: Text(
                  chatList[index].date,
                  textAlign: TextAlign.center,
                  style: TextStyle(
                      fontSize: gropByTextsize,
                      fontWeight: FontWeight.bold,
                      color: groupByTextColor),
                ),
              ),
              ChatBubble(
                message: chatList[index].message.toString(),
                isSender: chatList[index].isSender,
                bubbleBorder: bubbleBorder,
                senderBgColor: senderBgColor,
                receiverBgColor: receiverBgColor,
                senderTextColor: senderTextColor,
                receiverTextColor: receiverTextColor,
              ),
            ],
          );
        });
  }
}


lib/widget/chatbubble.dart
import 'package:flutter/material.dart';

class ChatBubble extends StatelessWidget {
  const ChatBubble(
      {super.key,
      required this.message,
      required this.isSender,
      required this.bubbleBorder,
      required this.receiverBgColor,
      required this.senderBgColor,
      required this.receiverTextColor,
      required this.senderTextColor});

  final String message;
  final bool isSender;
  final double bubbleBorder;
  final Color receiverBgColor, receiverTextColor;
  final Color senderBgColor, senderTextColor;

  @override
  Widget build(BuildContext context) {
    return Padding(
      // asymmetric padding
      padding: EdgeInsets.fromLTRB(
        isSender ? 64.0 : 16.0,
        4,
        isSender ? 16.0 : 64.0,
        4,
      ),
      child: Align(
        // align the child within the container
        alignment: isSender ? Alignment.centerRight : Alignment.centerLeft,
        child: DecoratedBox(
          // chat bubble decoration
          decoration: BoxDecoration(
            color: isSender ? senderBgColor : receiverBgColor,
            borderRadius: BorderRadius.circular(bubbleBorder),
          ),
          child: Padding(
            padding: const EdgeInsets.all(12),
            child: Text(
              message,
              style: TextStyle(
                  color: isSender ? senderTextColor : receiverTextColor),
            ),
          ),
        ),
      ),
    );
  }
}


example/lib/ChatPageScreen.dart
import 'package:flutter/material.dart';
import 'package:flutter_chat_page_design/flutter_chat.dart';
import 'package:flutter_chat_page_design/model/chatmodel.dart';

import 'customdata/customdata.dart';

class ChatPage extends StatefulWidget {
  const ChatPage({super.key});

  @override
  State<ChatPage> createState() => _ChatPageState();
}

class _ChatPageState extends State<ChatPage> {
  CustomData customData =  CustomData();
  List<ChatModel> chatList = [];

  @override
  void initState() {

    super.initState();
    chatList = customData.setData();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
          child: ChatListview(
        chatList: chatList,
        isDategroup: true,
        isReverse: true,
        gropByTextsize: 12,
        bubbleBorder: 15,

      )),
    );
  }
}


example/lib/customdata/customdata.dart

import 'package:flutter_chat_page_design/model/chatmodel.dart';

class CustomData {
  List chatList = [
    {'message': 'Hello', 'isSender': false, 'date': '22-08-2023'},
    {'message': 'Hi', 'isSender': true, 'date': '22-08-2023'},
    {'message': 'How are you?', 'isSender': true, 'date': '22-08-2023'},
    {'message': 'Good', 'isSender': false, 'date': '23-08-2023'},
    {'message': 'Nice', 'isSender': true, 'date': '23-08-2023'},
    {'message': 'Yupp', 'isSender': false, 'date': '23-08-2023'},
    {'message': 'any thing else', 'isSender': true, 'date': '24-08-2023'},
    {'message': 'No', 'isSender': false, 'date': '24-08-2023'},
    {'message': 'Really', 'isSender': true, 'date': '24-08-2023'},
    {'message': 'I cannot bealive', 'isSender': true, 'date': '24-08-2023'},
    {'message': 'No', 'isSender': false, 'date': '24-08-2023'},
    {'message': 'Okay', 'isSender': true, 'date': '24-08-2023'},
    {'message': 'where are you', 'isSender': true, 'date': '24-08-2023'}
  ];
  List<ChatModel> chatListnew = [];
  ChatModel chatModel =  ChatModel();

  List<ChatModel> setData() {
    for(int i =0;i<chatList.length;i++){
      ChatModel chatModel =  ChatModel();
      chatModel.message=chatList.elementAt(i)['message'];
      chatModel.isSender=chatList.elementAt(i)['isSender'];
      chatModel.date=chatList.elementAt(i)['date'];
      chatListnew.add(chatModel);
    }

    return chatListnew.reversed.toList();
  }
}
